### PR TITLE
Fix 1530 and fix 1531

### LINKED ISSUE #1530: Feature: Re-implement `MakeOverviewMessage()` in Report
## Feature Details
Update `MakeOverviewMessage()` from `ReportHelpers.R` to run with new Reporting framework. It should take the needed data sets as inputs instead of a `snapshot` object. Should return (roughly) the same text as in v1. Rename function following new convention as `Report_OverviewText()`

### LINKED ISSUE #1531: Bugfix: Troubleshoot Show/Hide functionality in `Report_StudyInfo`
## Expected Behavior
Table should show/hide based on the toggle created in `Report_StudyInfo` in `dev-v2` branch. 

## Current Behavior
Full table shown on initial click, but no further updates on subsequent clicks. 

## Steps to Reproduce

1.
2.
3.
4.

### PR DESCRIPTION
**Pull request by taylorrodgers**
## 1530 
Fix #1530 was to add the overview table summary below the table. 

This fix required the following changes:
* `Report_KRI.R` to include `strType` as an argument, which must be `country `or `site`
* `Report_OverviewText.R` to utilize the new data structure to generate the text
* `inst/report/Report_KRI.Rmd` to use the new `Report_OverviewText()` and to use `strType` as a param

## Test Notes/Sample Code
You can view the new output with the code found [here](https://github.com/Gilead-BioStats/gsm/pull/1525), or you can see in the screenshots below proof that it's there for both **site** and **country**, respectively.

image

image

## 1531
Fix #1531 was to fix the toggle for the Study Status summary. 

This fix required altering `inst/utils-0.0.1/showMetaTableDetails.js` to not alter a div with the id of `timeline`. 

You can see in the gif below that the error is no longer present:

https://github.com/Gilead-BioStats/gsm/assets/76186904/66e26994-d0d9-4c61-9293-0b82d87faca4

### CODE REVIEW

**Comment by taylorrodgers** [comment]
@jwildfire, this is ready for your review. One thing to note: I was unable to re-document the package. There are some conflict issues with the roxygen on my Atorus system versus the required package version for gsm.

File: R/Report_KRI.R [hunk 1]
```diff
@@ -13,19 +13,20 @@
...
 
 Report_KRI <- function(
   lCharts = NULL,
   dfSummary = NULL,
   dfStudy = NULL,
   dfSite = NULL,
   dfMetrics = NULL,
-  strOutpath = NULL
+  strOutpath = NULL,
+  strType = NULL
```
Comment (jwildfire) [inline R/Report_KRI.R:25]: I don't think this is needed. See the `group` derivation in `Report_Setup()` instead.

File: R/Report_OverviewText.R [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
-#   } else if (report == "country") {
-#     cat(glue::glue("<div>As of {status_study$gsm_analysis_date}, {status_study$studyid} has {round(sum(as.numeric(overview_raw_table$Subjects)))} participants enrolled across
-# {nrow(overview_raw_table)} countries. {red_kris} Country-KRI combinations have been flagged as red across {overview_raw_table %>% filter(.data$`Red KRIs` != 0) %>% nrow()} countries as shown in the Study Overview Table above.</div>
-#   - <div>{overview_raw_table %>% filter(.data$`Red KRIs` != 0) %>% nrow()} countries have at least one red KRI</div>
-#   - <div>{overview_raw_table %>% filter(.data$`Red KRIs` != 0 | .data$`Amber KRIs` != 0) %>% nrow()} countries have at least one red or amber KRI</div>
-#   - <div>{overview_raw_table %>% filter(.data$`Red KRIs` == 0 & .data$`Amber KRIs` == 0) %>% nrow()} countries have neither red nor amber KRIs</div>"), sep = "\n")
-#   }
-}
\ No newline at end of file
+Report_OverviewText <- function(strType, lSetup, dfSummary, dfStudy) {
```
Comment (jwildfire) [inline R/Report_OverviewText.R:8]: Remove `strType` and use `lSetup$group` instead.

**Review by jwildfire — CHANGES_REQUESTED** [review]

**Comment by jwildfire** [comment]
Looks good in general @taylorrodgers. Let's update the approach to `strType` from comments above and then I can re-review/merge.

File: R/Report_KRI.R (hunk 1 above)
Comment (taylorrodgers) [reply R/Report_KRI.R:25]: @jwildfire , this should be complete now on the most recent commit.

File: R/Report_OverviewText.R (hunk 2 above)
Comment (taylorrodgers) [reply R/Report_OverviewText.R]: @jwildfire , these changes are completed. I had to merge some changes that were not on my local from v2-dev, so it'll look more in-line with what you had changed last week. Some of these required some conflict resolutions in the code.

**Review by jwildfire — APPROVED** [review]

**Pull request merged**